java-tools/OsmAndServer/src/main/java/net/osmand/server/controllers/user/MapApiController.java

```diff
@@ -833,4 +828,112 @@ public String getRegionsByLatlon(@RequestParam("lat") double lat, @RequestParam(
+		if (userFile != null) {
+			boolean saved = userdataService.saveAccessedUser(dev, userFile);
+			if (!saved) {
+				return ResponseEntity.badRequest().body("User is in the blacklist");
+			}
+			if (downloadFile != null && downloadFile) {
+				FileDownloadResult fileResult = userdataService.getFile(userFile, dev);
+				if (fileResult == null) {
+					return ResponseEntity.badRequest().body(FILE_NOT_FOUND);
+				}
+				return ResponseEntity.ok()
+						.header("Content-Disposition", "attachment; filename=" + fileResult.fileName)
+						.contentType(org.springframework.http.MediaType.valueOf(fileResult.contentType))
+						.body(new InputStreamResource(fileResult.inputStream));
+			}
+
+			GpxFile gpxFile = userdataService.getFile(userFile);
+
+			if (gpxFile.getError() == null) {
+				GpxTrackAnalysis analysis = gpxFile.getAnalysis(System.currentTimeMillis());
+				WebGpxParser.TrackData gpxData = gpxService.getTrackDataByGpxFile(gpxFile, null, analysis);
+				if (gpxData != null) {
+					return ResponseEntity.ok(gsonWithNans.toJson(Map.of("gpx_data", gpxData)));
+				}
+			}
+		}
+		return ResponseEntity.badRequest().body(FILE_NOT_FOUND);
+	}
+
+	public record FileDownloadResult(InputStream inputStream, String fileName, String contentType) {
+	}
+
+	@GetMapping(path = {"/accessed-users"}, produces = "application/json")
+	public ResponseEntity<String> getAccessedUsers(@RequestParam String name,
+	                                               @RequestParam String type) {
+		PremiumUserDevicesRepository.PremiumUserDevice dev = checkUser();
+		if (dev == null) {
+			return tokenNotValid();
+		}
+		PremiumUserFilesRepository.UserFile userFile = userdataService.getUserFile(name, type, null, dev);
+		if (userFile == null) {
+			return ResponseEntity.badRequest().body(FILE_NOT_FOUND);
+		}
+		List<String> users = userdataService.getAccessedUsers(userFile);
+		return ResponseEntity.ok(gson.toJson(Map.of("accessedUsers", users)));
+	}
+
+	@PostMapping(path = {"/create-file-blacklist"}, produces = "application/json")
+	public ResponseEntity<String> createFileBlacklist(@RequestBody List<String> list,
+	                                                  @RequestParam String name,
+	                                                  @RequestParam String type) {
+		PremiumUserDevicesRepository.PremiumUserDevice dev = checkUser();
+		if (dev == null) {
+			return tokenNotValid();
+		}
+		PremiumUserFilesRepository.UserFile userFile = userdataService.getUserFile(name, type, null, dev);
+		if (userFile == null) {
+			return ResponseEntity.badRequest().body(FILE_NOT_FOUND);
+		}
+		boolean created = userdataService.createFileBlacklist(userFile, list);
+		if (!created) {
+			return ResponseEntity.badRequest().body("Error creating blacklist");
+		}
+		return ResponseEntity.ok("Blacklist created");
+	}
+
+	@GetMapping(path = {"/get-blacklist"}, produces = "application/json")
```

Comment:
Method get file shared information is enough with all information of json. No secrets could be to the owner of the file.

Reply:
I don't understand what you mean. This is api for showing the list to the user.